tests/block-creation.spec.js

```diff
@@ -0,0 +1,60 @@
+import {expect, test} from "@playwright/test";
+import {HeynotePage} from "./test-utils.js";
+
+let heynotePage
+
+test.beforeEach(async ({ page }) => {
+    console.log("beforeEach")
+    heynotePage = new HeynotePage(page)
+    await heynotePage.goto()
+
+    expect((await heynotePage.getBlocks()).length).toBe(1)
+    heynotePage.setContent(`
+∞∞∞text
+Block A
+∞∞∞text
+Block B
+∞∞∞text
+Block C`)
+    // check that blocks are created
+    expect((await heynotePage.getBlocks()).length).toBe(3)
+
+    // check that visual block layers are created
+    expect(await page.locator("css=.heynote-blocks-layer > div").count()).toBe(3)
+
+    // select the second block
+    await page.locator("body").press("ArrowUp")
+});
+
+test("create block before current", async ({ page }) => {
+    await runTest(page, "Alt+Enter", ['A', 'D', 'B', 'C'])
+})
+
+test("create block after current", async ({ page }) => {
+    await runTest(page, heynotePage.isMac ? "Meta+Enter" : "Control+Enter",['A', 'B', 'D', 'C'])
+})
+
+test("create block before first", async ({ page }) => {
+    await runTest(page, "Alt+Shift+Enter",['D', 'A', 'B', 'C'])
+})
+
+test("create block after last", async ({ page }) => {
+    await runTest(page, heynotePage.isMac ? "Meta+Shift+Enter" : "Control+Shift+Enter",['A', 'B', 'C', 'D'])
+})
+
+const runTest = async (page, key, expectedBlocks) => {
+    // create a new block
+    await page.locator("body").press(key)
+    await page.waitForTimeout(100);
+    await page.locator("body").pressSequentially("Block D")
```

Comment:
In my _personal_ view, runTest moves too much logic out from each test case, making it less clear what they test. **But I do not have much to do with this repo so my views don't mean much here**. I'm just sharing some unsolicited opinions here.

This code pressing buttons and typing into the blocks I think would be fine to repeat in each test case.

And the expect code below could in my opinion be more readable if the `expect` calls were in each test case, but to get rid of some repetition you could introduce a helper `getAllBlockContents()` that just returns an array of all the contents from the blocks, and in the test case code you can test `expect(getAllBlockContents()).toBe(["Block A", "Block B", "Block D", "Block C"])` etc. (I am assuming the test framework can compare arrays of strings like this, it's been a while since I did JavaScript so I may be misremembering).

Reply:
Eh, there is already a test case dedicated to this in [editing.spec.js](https://github.com/heyman/heynote/blob/main/tests/basic-editing.spec.js), which is why I abstracted it as much. I consider that filling the contents of a block is already tested/working, which is not the scope of this test. The point here is to focus on the keybind and the expected outcome for the blocks.